SARS-CoV-2-Mpro/consolidate-work-to-pandas.py

```diff
@@ -0,0 +1,173 @@
+#!/usr/bin/env python
+
+"""
+Extract final work values to a single pandas dataframe
+"""
+
+import os
+
+projects = {
+    'complex' : 13418,
+    'solvent' : 13419,
+    }
+
+NUM_EXPECTED_WORK = 4001
+NUM_EXPECTED_WORK = 401
+NUM_EXPECTED_WORK = 41
+
+phases = list(projects.keys())
+
+basepaths = [
+    '/home/server/server2/data/SVR314342810/',
+    ]
+
+import datetime
+timestamp = datetime.datetime.now()
+
+def get_results(PROJ):
+    """
+    Get all result directories for a given project
+
+    Parameters
+    ----------
+    PROJ : str
+        PROJ, such as 'PROJ13400'
+
+    Returns
+    -------
+    results : dict
+        results[path] = (PROJ,RUN,CLONE,GEN) where path is the results directory
+    """
+    from tqdm import tqdm
+    
+    # Make a list of all results packets paths from all locations
+    paths = list()
+    from glob import glob # glob takes forever on S3 file gateway, so we don't use it
```

Comment:
Not sure about this since I'm unfamiliar with S3 file gateway, but it might be more efficient and/or require less code to use the S3 API directly here. E.g. https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/s3.html#S3.Client.list_objects_v2